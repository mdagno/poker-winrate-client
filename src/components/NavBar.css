.hamburger-menu {
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  text-align: right;
  margin-right: 10%;
  z-index: 10;
}

.hamburger-menu i {
  color: white;
  font-size: 1.25em;
}

.navBar i:hover{
  color:  white;
}

.navLinks {
  list-style: none;
  padding: 0;
  background-color: white;
  text-align: right;
  width: 150px;
  z-index: 10;
  color: #222;
  box-shadow: 3px 3px 3px #777;
}

.navBar li {
  padding: 0.5em;
  padding-right: 0.8em;
  margin-top: 0px;
  font-size: 1em;
  font-style: italic;
  font-weight: 500;
}

.one {
  background-color: #ddd;
}

.two {
  background-color: white;
}

.navBar a {
  text-decoration: none;
  color: black;
}

.navBar li:hover{
  background-color: #00866F; 
}

