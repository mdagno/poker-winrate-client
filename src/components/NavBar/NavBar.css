.hamburger-menu {
  display: flex;
  flex-direction: row;
  justify-content: flex-end;
  text-align: right;
  margin-right: 10%;
  z-index: 10;
}

.hamburger-menu i {
  color: white;
  font-size: 1.25em;
}

.navBar i:hover{
  color:  white;
}

.navLinks {
  list-style: none;
  padding: 0;
  background-color: white;
  text-align: right;
  width: 150px;
  z-index: 10;
  color: #222;
  box-shadow: 3px 3px 3px #777;
}

.navBar li {
  padding: 0.5em;
  padding-right: 0.8em;
  margin-top: 0px;
  font-size: 1em;
  font-weight: 500;
}

.one {
  background-color: #ddd;
}

.two {
  background-color: white;
}

.navBar a {
  text-decoration: none;
  color: black;
}

.navBar li:hover{
  background-color: #00866F; 
}

.navLinksDesktop {
  display: none;
}

@media only screen and (min-device-width : 768px){
  .hamburger-menu {
    display: none;
  }

  .navLinksDesktop {
    list-style: none;
    display: flex;
    justify-content: center;
    width: auto;
    background-color:  #475080;
    padding: 1em;
    border-bottom: 2px solid white;
  }


  .navLinksDesktop li{
    text-align: center;
    font-size: 0.9em;
    margin-left: 1.5em;
    margin-top: 0.5em;
    width: 7.7em;
    border-radius: 5px;
    border: 1px white outset;
  }

  .navLinksDesktop li:hover {
    width: 8.2em;
  }
}
