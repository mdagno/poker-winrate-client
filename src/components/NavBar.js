import React from 'react';
import { NavLink } from 'react-router-dom';
import './NavBar.css'

export default class NavBar extends React.Component{

renderNavRoutes = () => {

}

  render(){
    return(
    <nav className='navBar'>
      <ul>
      <li>
        <NavLink to='/'>
          Summary
        </NavLink>
        </li>
        <li>
        <NavLink to='/sessions'>
          Your Sessions
        </NavLink>
        </li>
        <li>
        <NavLink to='/addsession'>
          Add a session
        </NavLink>
        </li>
        <li>
        <NavLink to='/analysis'>
          Analytics
        </NavLink>
        </li>
      </ul>
    </nav>
    )
  }
}