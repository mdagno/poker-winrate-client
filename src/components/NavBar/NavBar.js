import React from 'react';
import { NavLink } from 'react-router-dom';
import './NavBar.css'
import ApiContext from '../../ApiContext';
import TokenService from '../../services/token-service';

export default class NavBar extends React.Component{
  state = {
    expandedView: false,
  }
  
  static contextType = ApiContext;

  handleHamburger = () => {
    this.setState({
      expandedView: !this.state.expandedView
    })
  }

  renderHamburgerMenu = () => {
    let navLinks;
    if (this.state.expandedView === true && TokenService.hasAuthToken()){
      navLinks = (
      <ul className='navLinks' onClick={this.handleHamburger}>
        <NavLink to='/home'>
        <li className="navLink one">
          Summary
        </li>
        </NavLink>
        <NavLink to='/sessions'>
        <li className="navLink two">
          Your Sessions
        </li>
        </NavLink>
        <NavLink to='/addsession'>
        <li className="navLink one">
          Add a session
        </li>
        </NavLink>
        <NavLink to='/analysis'>
        <li className="navLink two">
          Analytics
        </li>
        </NavLink>
      </ul>
      )
    }
    return navLinks;
  }

  renderNavLinks = () => {
    if(TokenService.hasAuthToken())
    return (
      <ul className="navLinksDesktop">
        <NavLink to='/home'>
        <li className="navLink one">
          Summary
        </li>
        </NavLink>
        <NavLink to='/sessions'>
        <li className="navLink two">
          Your Sessions
        </li>
        </NavLink>
        <NavLink to='/addsession'>
        <li className="navLink one">
          Add a session
        </li>
        </NavLink>
        <NavLink to='/analysis'>
        <li className="navLink two">
          Analytics
        </li>
        </NavLink>
        <NavLink to='/'>
        <li className="navLink two" onClick={() => this.context.processLogout()}>
          Logout
        </li>
        </NavLink>
    </ul>
    )
    else {
      return (
        <ul className="navLinksDesktop">
        <NavLink to='/'>
        <li className="navLink one">
          Home
        </li>
        </NavLink>
        <NavLink to='/login'>
        <li className="navLink two">
          Login
        </li>
        </NavLink>
        <NavLink to='/register'>
        <li className="navLink one">
          Register
        </li>
        </NavLink>
    </ul>
      )
    }
  }

  render(){
    return(
    <nav className='navBar'>
      <div className='hamburger-menu'>
      <i class="fas fa-bars" onClick={this.handleHamburger}></i>
      {window.innerWidth < 768 ? this.renderHamburgerMenu() : null}
      </div>
      {this.renderNavLinks()}
    </nav>
    )
  }
}