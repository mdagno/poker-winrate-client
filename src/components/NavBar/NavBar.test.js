import React from 'react';
import NavBar from './NavBar';
import renderer from 'react-test-renderer';
import ReactDOM from 'react-dom';
import { BrowserRouter } from 'react-router-dom';

describe('NavBar component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(      
    <BrowserRouter>
      <NavBar />
     </BrowserRouter>, div);
    ReactDOM.unmountComponentAtNode(div);
  });

  it('NavBar renders correctly', () => {
    const tree = renderer
      .create(    
      <BrowserRouter>
        <NavBar />
      </BrowserRouter>)
      .toJSON();
    expect(tree).toMatchSnapshot();  
  });
})


