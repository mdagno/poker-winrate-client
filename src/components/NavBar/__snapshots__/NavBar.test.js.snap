// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavBar component NavBar renders correctly 1`] = `
<nav
  className="navBar"
>
  <div
    className="hamburger-menu"
  >
    <i
      class="fas fa-bars"
      onClick={[Function]}
    />
  </div>
  <ul
    className="navLinksDesktop"
  >
    <a
      aria-current="page"
      className="active"
      href="/"
      onClick={[Function]}
      style={Object {}}
    >
      <li
        className="navLink one"
      >
        Home
      </li>
    </a>
    <a
      aria-current={null}
      href="/login"
      onClick={[Function]}
    >
      <li
        className="navLink two"
      >
        Login
      </li>
    </a>
    <a
      aria-current={null}
      href="/register"
      onClick={[Function]}
    >
      <li
        className="navLink one"
      >
        Register
      </li>
    </a>
  </ul>
</nav>
`;
